sig-release/members.md

```diff
@@ -0,0 +1,8 @@
+## SIG Release Membership
```

Comment:
@calebamiles I'm ok with merging it as is - don't want this item to be a blocker.

But IMO, it's not a common practice for SIG's to have a similar list of people.